vumi/components/message_store.py

```diff
@@ -431,12 +440,18 @@ def add_event(self, event):
             event_record = self.events(event_id, event=event, message=msg_id)
         else:
             event_record.event = event
-        yield event_record.save()
 
-        msg_record = yield self.outbound_messages.load(msg_id)
-        if msg_record is not None:
-            for batch_id in msg_record.batches.keys():
-                yield self.cache.add_event(batch_id, event)
+        # If we aren't given batch_ids, get them from the outbound message.
+        if batch_ids is None:
+            msg_record = yield self.outbound_messages.load(msg_id)
+            if msg_record is not None:
```

Comment:
If it's been deleted or hasn't been stored yet. It's not a situation we expect, but we already handle it in the old code and there's no particular reason to change that.